Sources/Plasma/FeatureLib/pfGLPipeline/plGLEnumerate.cpp

```diff
@@ -0,0 +1,306 @@
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#include "HeadSpin.h"
+
+#include <epoxy/gl.h>
+#include <vector>
+
+#include "plPipeline/hsG3DDeviceSelector.h"
+
+bool fillDeviceRecord(hsG3DDeviceRecord& devRec)
+{
+    if (epoxy_gl_version() < 33)
+        return false;
+
+    const char* renderer = reinterpret_cast<const char*>(glGetString(GL_RENDERER));
+    devRec.SetDeviceDesc(renderer);
+
+    const char* vendor = reinterpret_cast<const char*>(glGetString(GL_VENDOR));
+    devRec.SetDriverDesc(vendor);
+
+    const char* version = reinterpret_cast<const char*>(glGetString(GL_VERSION));
+    devRec.SetDriverVersion(version);
+
+    devRec.SetCap(hsG3DDeviceSelector::kCapsMipmap);
+    devRec.SetCap(hsG3DDeviceSelector::kCapsPerspective);
+    devRec.SetCap(hsG3DDeviceSelector::kCapsCompressTextures);
+    devRec.SetCap(hsG3DDeviceSelector::kCapsDoesSmallTextures);
+
+    devRec.SetLayersAtOnce(8);
+
+    // Just make a fake mode so the device selector will let it through
+    hsG3DDeviceMode devMode;
+    devMode.SetWidth(hsG3DDeviceSelector::kDefaultWidth);
+    devMode.SetHeight(hsG3DDeviceSelector::kDefaultHeight);
+    devMode.SetColorDepth(hsG3DDeviceSelector::kDefaultDepth);
+    devRec.GetModes().emplace_back(devMode);
+
+    return true;
+}
+
+
+#ifdef USE_EGL
+#include <epoxy/egl.h>
+
+void plEGLEnumerate(std::vector<hsG3DDeviceRecord>& records)
+{
+    EGLDisplay display = EGL_NO_DISPLAY;
+    EGLContext context = EGL_NO_CONTEXT;
+    EGLSurface surface = EGL_NO_SURFACE;
+
+    do {
+        display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+        if (display == EGL_NO_DISPLAY)
+            break;
+
+        if (!eglInitialize(display, nullptr, nullptr))
+            break;
+
+        if (!eglBindAPI(EGL_OPENGL_API))
+            break;
+
+        GLint numConfigs = 0;
+        if (!eglGetConfigs(display, nullptr, 0, &numConfigs) || numConfigs == 0)
+            break;
+
+        std::vector<EGLConfig> configs(numConfigs);
+        if (!eglGetConfigs(display, configs.data(), configs.size(), &numConfigs))
+            break;
+
+        EGLint ctx_attrs[] = {
+            EGL_CONTEXT_MAJOR_VERSION, 3,
+            EGL_CONTEXT_OPENGL_PROFILE_MASK, EGL_CONTEXT_OPENGL_CORE_PROFILE_BIT,
+            EGL_NONE
+        };
+
+        context = eglCreateContext(display, configs[0], EGL_NO_CONTEXT, ctx_attrs);
+        if (context == EGL_NO_CONTEXT)
+            break;
+
+        surface = eglCreatePbufferSurface(display, configs[0], nullptr);
+        if (surface == EGL_NO_SURFACE)
+            break;
+
+        if (!eglMakeCurrent(display, surface, surface, context))
+            break;
+
+        hsG3DDeviceRecord devRec;
+        devRec.SetG3DDeviceType(hsG3DDeviceSelector::kDevTypeOpenGL);
+        devRec.SetDriverName("EGL");
+
+        if (fillDeviceRecord(devRec))
+            records.emplace_back(devRec);
```

Comment:
```suggestion
            records.push_back(devRec);
```

Reply:
`emplace_back` matches what the DX side is doing: https://github.com/H-uru/Plasma/blob/f7811a2f3728eb9206d194f3615d876a82400448/Sources/Plasma/FeatureLib/pfDXPipeline/plDXEnumerate.cpp#L670